x-pack/plugins/security_solution/public/detections/components/alerts_kpis/alerts_by_type_panel/alerts_by_type.test.tsx

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { act, render } from '@testing-library/react';
+import React from 'react';
+import { TestProviders } from '../../../../common/mock';
+import { AlertsByType } from './alerts_by_type';
+import { parsedAlerts } from './mock_data';
+
+jest.mock('../../../../common/lib/kibana');
+
+jest.mock('react-router-dom', () => {
+  const actual = jest.requireActual('react-router-dom');
+  return { ...actual, useLocation: jest.fn().mockReturnValue({ pathname: '' }) };
+});
+
+describe('Alert by type chart', () => {
+  const defaultProps = {
+    data: [],
+    isLoading: false,
+  };
+
+  afterEach(() => {
+    jest.clearAllMocks();
+    jest.restoreAllMocks();
+  });
+
+  test('renders health and pallette display correctly without data', () => {
+    act(() => {
+      const { container } = render(
+        <TestProviders>
+          <AlertsByType {...defaultProps} />
+        </TestProviders>
+      );
+      expect(
+        container.querySelector('[data-test-subj="alerts-by-type-palette-display"]')
```

Comment:
To simplify this, I would just pull out the selector strings to constants and re-use that. For example: `const alertsByType = '[data-test-subj="alerts-by-type-palette-display"]'` or a better name